charts/alfresco-search-enterprise/templates/_helpers-elasticsearch.tpl

```diff
@@ -1,5 +1,12 @@
 {{- define "spring.elasticsearch.config" -}}
-  SPRING_ELASTICSEARCH_REST_URIS: "{{ .Values.elasticsearch.protocol | default .Values.global.elasticsearch.protocol }}://{{ .Values.elasticsearch.host | default .Values.global.elasticsearch.host }}:{{ .Values.elasticsearch.port | default .Values.global.elasticsearch.port }}"
+{{- if .Values.elasticsearch.enabled }}
+  SPRING_ELASTICSEARCH_REST_URIS: "{{ .Values.elasticsearch.protocol }}://{{ .Values.elasticsearch.host }}:{{ .Values.elasticsearch.port }}"
+{{- else }}
+  {{- if or (not .Values.global.elasticsearch.protocol) (or (not .Values.global.elasticsearch.host) (not .Values.global.elasticsearch.port)) }}
+    {{ fail "Please provide external elasticsearch details under global.elasticsearch or enable the embedded elasticsearch via elasticsearch.enabled" }}
+  {{- end }}
+  SPRING_ELASTICSEARCH_REST_URIS: "{{ .Values.global.elasticsearch.protocol  }}://{{ .Values.global.elasticsearch.host }}:{{ .Values.global.elasticsearch.port }}"
```

Comment:
this come out quite naturally: before we had the `elasticsearch` dependency defined in the main chart, while now it's inside this chart and whatever I put under `.Values.elasticsearch` is meant to configure the external `elasticsearch` chart and continue to be an override for `.Values.global.elasticsearch` seems confusing to me.

For me is enough to have only the `.Values.global.elasticsearch` to specify external elasticsearch details, that for sure we need to have because it will be the only way to define things in a single place and use them in multiple charts.